InterfaceFortran/TasmanianSG.f90

```diff
@@ -154,501 +166,646 @@ Module TasmanianSG
-FUNCTION tsgIsSetDomainTransfrom(gridID) result(res)
-  INTEGER :: gridID, res
-  CALL tsgidt(gridID, res)
-END FUNCTION tsgIsSetDomainTransfrom
-!=======================================================================
-SUBROUTINE tsgClearDomainTransform(gridID)
-  INTEGER :: gridID
-  CALL tsgcdt(gridID)
-END SUBROUTINE tsgClearDomainTransform
-!=======================================================================
-SUBROUTINE tsgGetDomainTransform(gridID, transformA, transformB)
-  INTEGER :: gridID
-  DOUBLE PRECISION :: transformA(*), transformB(*)
-  CALL tsggdt(gridID, transformA, transformB)
-END SUBROUTINE tsgGetDomainTransform
-!=======================================================================
-SUBROUTINE tsgSetAnisotropicRefinement(gridID, gtype, minGrowth, output)
-  INTEGER :: gridID, gtype, minGrowth, output
-  CALL tsgsar(gridID, gtype, minGrowth, output)
-END SUBROUTINE tsgSetAnisotropicRefinement
-!=======================================================================
-FUNCTION tsgEstimateAnisotropicCoefficients(gridID, gtype, output) &
-                                              result(coeff)
-  INTEGER :: gridID, gtype, output, N
-  INTEGER, allocatable :: coeff(:)
+  allocate(w(length))
+  call tsggiw(gridID, x, w)
+end function tsgGetInterpolationWeights
+!=======================================================================
+subroutine tsgGetInterpolationWeightsStatic(gridID, x, weights)
+  integer :: gridID
+  double precision :: x(:)
+  double precision :: weights(:)
+  call tsggiw(gridID, x, weights)
+end subroutine tsgGetInterpolationWeightsStatic
+!=======================================================================
+subroutine tsgLoadNeededPoints(gridID, values)
+  integer :: gridID
+  double precision :: values(:,:)
+  call tsglnp(gridID, values)
+end subroutine tsgLoadNeededPoints
+!=======================================================================
+subroutine tsgEvaluate(gridID, x, y)
+  integer :: gridID
+  double precision, intent(in) :: x(:)
+  double precision :: y(:)
+  call tsgeva(gridID, x, y)
+end subroutine tsgEvaluate
+!=======================================================================
+subroutine tsgEvaluateFast(gridID, x, y)
+  integer :: gridID
+  double precision, intent(in) :: x(:)
+  double precision :: y(:)
+  call tsgevf(gridID, x, y)
+end subroutine tsgEvaluateFast
+!=======================================================================
+subroutine tsgEvaluateBatch(gridID, x, numX, y)
+  integer :: gridID, numX
+  double precision :: x(:,:), y(:,:)
+  call tsgevb(gridID, x, numX, y)
+end subroutine tsgEvaluateBatch
+!=======================================================================
+subroutine tsgEvaluateHierarchicalFunctions(gridID, x, numX, y)
+  integer :: gridID, numX
+  double precision :: x(:,:), y(:,:)
+  double precision :: y_c_style(size(y,2),size(y,1))
+  call tsgehf(gridID, x, numX, y_c_style)
```

Comment:
* if the grid is not Fourier, then call the C function, otherwise print an error message 
```
ERROR: called tsgEvaluateHierarchicalFunctions() on a Fourier grid, use tsgEvaluateComplexHierarchicalFunctions() instead
```
We will deal with rigorous error-checking and message passing later.